src/vm/comtoclrcall.cpp

```diff
@@ -792,13 +787,22 @@ extern "C" UINT64 __stdcall COMToCLRWorker(Thread *pThread, ComMethodFrame* pFra
         retVal = hr;
     }
     else if (pCMD->IsNativeBoolRetVal())
-        retVal = 0;
+    {
+        retVal = FALSE;
+    }
     else if (pCMD->IsNativeR4RetVal())
-        setFPReturn(4, CLR_NAN_32);
+    {
+        setFPReturn(sizeof(CLR_NAN_32), CLR_NAN_32);
+    }
     else if (pCMD->IsNativeR8RetVal())
-        setFPReturn(8, CLR_NAN_64);
+    {
+        setFPReturn(sizeof(CLR_NAN_64), CLR_NAN_64);
```

Comment:
This expand to `sizeof(IA64(0xFFF8000000000000))`. I guess it still returns 8, but I do not think this is improving readability. If `CLR_NAN_64` constant happened to have different value, it can very well be wrong.

Reply:
That is true, but if that size changes then it can't be a nan value since it is a defined bit pattern in IEEE. This constant (size and value) cannot change or else it won't be a double nan.

Reply:
I am fine putting it back though.